extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/filter/OperationFilter.java

```diff
@@ -97,6 +100,21 @@ private String methodRef(Operation operation) {
         return (String) (extensions != null ? extensions.get(EXT_METHOD_REF) : null);
     }
 
+    private void maybeAddBadRequestResponse(Map.Entry<PathItem.HttpMethod, Operation> operation, String methodRef) {
+        if (!classNameMap.containsKey(methodRef)) {
+            return;
+        }
+
+        if (doAutoBadRequest && (operation.getKey().equals(PathItem.HttpMethod.POST)
+                || operation.getKey().equals(PathItem.HttpMethod.PUT)) && operation.getValue().getRequestBody() != null) { // Only applies to input
```

Comment:
Can you remind me the type of `getRequestBody`?

Reply:
org.eclipse.microprofile.openapi.annotations.parameters.RequestBody

Reply:
Ok, so you are sure that when this is not null there is a body?

Reply:
I am sure, but let's ask @MikeEdgar for input, I might be missing something.

Reply:
I think a non-null requestBody should be safe. That object also contains a required `content` attribute (per OAS), so the presence of this object should be enough.

Question - should this also check for `PATCH` requests?

Also, I wonder if simple cases such as `String` or file request bodies, this may not make sense since there wouldn't be deserialization involved.

Reply:
Good point. Let me see what Quarkus REST does with PATCH and String / File, and make sure that openapi is synced with that.

Reply:
Ok Quarkus REST for PATCH does not have the same default as POST and PUT (400 - Bad Request). So for now let's only do those two. I also added to check for String, Numbers, Boolean and File, and in those cases not add the default. I think we are good to go now.